src/commands/CountryCommand.ts

```diff
@@ -0,0 +1,391 @@
+			}
+			case "specific": {
+				const country: Country | undefined = getCountryByName(interaction.options.getString("country") ?? "Bhutan");
+				const info: string = interaction.options.getString("info") ?? "population";
+
+				if (country == undefined) {
+					countryUndefinedReply(interaction);
+				} else {
+					interaction.reply(specificRequestReplies[info](country));
+				}
+				break;
+			}
+			case "query": {
+				const sortCriteria: CountryKey = (interaction.options.getString("sort-criteria") ?? "none") as CountryKey;
+				const order = interaction.options.getString("order") ?? "ascending";
+				const scale = interaction.options.getInteger("scale") ?? 10;
+				const filterCriteria: CountryKey = (interaction.options.getString("filter-criteria") ?? "none") as CountryKey;
+				const relation = interaction.options.getString("relation") ?? "eq";
+				const filterValue = interaction.options.getString("filter-value") ?? "";
+				let includeData = interaction.options.getBoolean("include-data") ?? true;
+				let numbered = false;
+				let embedDataCriteria: CountryKey = sortCriteria;
+
+				// sorting
+				if ((sortCriteria as string) !== "none") {
+					numbered = true;
+					sortCountryDataBy(sortCriteria);
+					if (order === "descending") {
+						countryData.reverse();
+					}
+				} else {
+					countryData.sort(() => Math.random() - 0.5); // shuffle
+					includeData = false;
+				}
+
+				// filtering
+				let data: Country[] = countryData;
+				let val: number | string | boolean = filterValue;
+				if ((filterCriteria as string) !== "none") {
+					if (["true", "false"].includes(filterValue)) {
+						val = filterValue === "true";
+					} else if (!isNaN(+filterValue)) {
+						val = +filterValue;
+					}
+
+					if (
+						typeof val !== typeOfCountryProperty(filterCriteria) &&
+						!(typeof val === "string" && typeOfCountryProperty(filterCriteria) === "object")
+					) {
+						thatDoesntMakeSenseReply(interaction);
+						break;
+					}
+
+					includeData = true;
+					data = getFilteredCountryDataBy(filterCriteria, relation, val);
+
+					if (relation !== "eq") {
+						embedDataCriteria = filterCriteria;
+					}
+				}
+
+				// output
+				const title = `${scale > countryData.length ? "All" : `Top ${scale}`} countries ${
+					(sortCriteria as string) !== "none" ? `listed by ${sortCriteria} in ${order} order` : "shuffeled"
+				}${(filterCriteria as string) !== "none" ? `, ${filteringTitles[relation](filterValue, filterCriteria)}` : ""}`;
+				interaction.reply({
+					embeds: [
+						getListEmbed(countriesToEmbedForm(data.slice(0, scale), embedDataCriteria, includeData), title, numbered),
+					],
+				});
+				break;
+			}
+			default: {
+				interaction.reply("Sorry, I don't know what to do... qwq");
+				break;
+			}
+		}
+	}
+
+	register():
+		| SlashCommandBuilder
+		| SlashCommandSubcommandsOnlyBuilder
+		| Omit<SlashCommandBuilder, "addSubcommandGroup" | "addSubcommand"> {
+		return new SlashCommandBuilder()
```

Comment:
Just a side mention on this. Also applies to some other stuff. Typing might be unnecessary as the correct type can be either inferred or is inherited from the extended class.

Reply:
do you I should just get rid of the typing there completely?

Reply:
Not necessary, but can do

Reply:
@KommentatorForAll 
What do you think?

Reply:
we could call .build() and just return the REST json whatever discord command resolvable stuff. that would indeed make it easier

Reply:
Sounds reasonable, might be better to have another pr for that after this one

Reply:
I'll leave it like that for now